src/libraries/System.Diagnostics.PerformanceCounter/tests/InstanceDataTests.cs

```diff
@@ -124,8 +124,16 @@ public static void InstanceDataCollectionCollection_CopyTo()
 
         public static InstanceDataCollectionCollection GetInstanceDataCollectionCollection()
         {
-            PerformanceCounterCategory pcc =  Helpers.RetryOnAllPlatforms(() => new PerformanceCounterCategory("Processor"));
-            return Helpers.RetryOnAllPlatforms(() => pcc.ReadCategory());
+            PerformanceCounterCategory pcc = Helpers.RetryOnAllPlatforms(() => new PerformanceCounterCategory("Processor"));
+            return Helpers.RetryOnAllPlatforms(() =>
+            {
+                var idcc = pcc.ReadCategory();
+                if (idcc.Values.Count == 0)
+                {
+                    Assert.InRange(idcc.Values.Count, 1, int.MaxValue);
```

Comment:
Unsure if I understand the actual issue but can't you just `throw new Exception();` to cause a retry?

Reply:
It's also a little more self documenting to assert rather than simply throw